src/app/clusters/thermostat-server/thermostat-server-presets.cpp

```diff
@@ -402,6 +402,30 @@ CHIP_ERROR ThermostatAttrAccess::AppendPendingPreset(Thermostat::Delegate * dele
         return CHIP_IM_GLOBAL_STATUS(ConstraintError);
     }
 
+    // Before adding this preset to the pending presets, if the expected length of the pending presets' list
+    // exceeds the total number of presets supported, return RESOURCE_EXHAUSTED. Note that the preset has not been appended yet.
+
+    uint8_t numberOfPendingPresets = CountNumberOfPendingPresets(delegate);
+
+    // Increment number of pending presets by 1 to account for this preset.
+    uint8_t totalExpectedCount = numberOfPendingPresets + 1;
+
+    uint8_t numberOfPresetsSupported = delegate->GetNumberOfPresets();
+
+    if (numberOfPresetsSupported == 0)
+    {
+        ChipLogError(Zcl, "AppendPendingPreset: Failed to get NumberOfPresets");
+        return CHIP_IM_GLOBAL_STATUS(InvalidInState);
```

Comment:
Shouldn't this be resource exhausted? Or just chuck this and let the next check handle it? Boris felt that if the delegate returned "0" for a timeout in atomic write, we couldn't return InvalidInState because the spec didn't account for the possibility that the delegate says something nonsensical, which is the same problem here.

Reply:
yeah i was thinking about that. will return resourceexhausted

Reply:
done.